### ISSUE TITLE
Support 'screenshot window' input command

### ISSUE THREAD

**Issue by wantMoreSleep** [iina/iina#4505]
**What you want IINA to do:**
As titled.

**What IINA does currently:**
Nothing happened after I've pressed key that is bind to this command.

**Why you think this should be added:**
I want to take screenshot with timestamp baked in.

**Examples of other projects that have something similar:**
The original mpv player does this.

**Comment by low-batt** [comment]
From the mpv manual section for the [Screenshot command](https://mpv.io/manual/stable/#command-interface-screenshot-%3Cflags%3E):
> **screenshot \<flags>**
>
>    Take a screenshot.
>
>    Multiple flags are available (some can be combined with +):
> …
>    **\<window>**
>        Save the contents of the mpv window. Typically scaled, with OSD and subtitles. The exact behavior depends on the selected video output, and if no support is available, this will act like video.

If I'm understanding the issue correctly, the request is to be able to bind a key to `screenshot window` and have it work like it does in the mpv player.

If currently you bind a key to this command and then press it the IINA log file will contain the following errors:
```text
19:33:14.238 [iina][e] Return value -12 when executing key command screenshot window
19:33:14.238 [mpv][e] [cplayer] error: Taking screenshot failed.
```

The mpv log file shows:
```text
[  68.575][d][cplayer] Run command: screenshot, flags=73, args=[flags="window", legacy="unused"]
[  68.593][e][cplayer] Taking screenshot failed.
```

A lot of mpv functionality is available in IINA. The exception is functionality that is tied to mpv's UI. Likely that is the problem in this case.

As it is mpv that generates the screenshot to support this we'd have to figure out how to integrate such functionality into mpv, if that is even possible.

**Comment by wantMoreSleep** [comment]
Yup, what you described is my problem. Looks like I will have to be contend with MPV, if there's really no other approach in IINA to take screenshot with timestamp or any other OSD baked-in. Will revisit this project in the future, and thanks for the reply.

**Comment by low-batt** [comment]
It is possible to embed a timestamp in the video using a filter. If you do that then the timestamp will appear in a screenshot. A quick example…

In IINA I added a FFmpeg [drawtext](https://ffmpeg.org/ffmpeg-filters.html#drawtext-1) video filter:
```text
box=1:boxborderw=10:fontcolor=black:fontfile=Times.ttc:fontsize=128:text=%{pts}:x=512:y=96
```

And took a screenshot:

![issue-4505](https://github.com/iina/iina/assets/86170219/f09433da-57f0-4047-a253-22003d7dd722)

**Comment by low-batt** [comment]
Support was added in IINA 1.4.0-beta1.
